packit_service/worker/parser.py

```diff
@@ -722,52 +727,77 @@ def parse_distgit_event(event) -> Optional[DistGitEvent]:
         )
 
     @staticmethod
-    def parse_testing_farm_results_event(event) -> Optional[TestingFarmResultsEvent]:
-        """ this corresponds to testing farm results event """
-        pipeline_id: str = nested_get(event, "pipeline", "id")
-        if not pipeline_id:
-            return None
-
-        result: TestingFarmResult = TestingFarmResult(event.get("result"))
-        environment: str = nested_get(event, "environment", "image")
-        message: str = event.get("message")
-        log_url: str = event.get("url")
-        copr_repo_name: str = nested_get(event, "artifact", "copr-repo-name")
-        copr_chroot: str = nested_get(event, "artifact", "copr-chroot")
-        repo_name: str = nested_get(event, "artifact", "repo-name")
-        repo_namespace: str = nested_get(event, "artifact", "repo-namespace")
-        ref: str = nested_get(event, "artifact", "git-ref")
-        https_url: str = nested_get(event, "artifact", "git-url")
-        commit_sha: str = nested_get(event, "artifact", "commit-sha")
-        tests: List[TestResult] = [
+    def _parse_tf_result_xunit(xunit: Optional[str]) -> List[TestResult]:
+        """ Parse event["result"]["xunit"] to get tests results """
+        if not xunit:
+            return []
+        xunit_dict = xmltodict.parse(xunit)
```

Comment:
I'd say that a more proper reaction would be
![](https://media1.tenor.com/images/ecdb4f38f98a7f403bfc441fb9603e83/tenor.gif)

Reply:
yes, xunit, which is xml, see for example ["result"]["xunit"] [here](https://api.dev.testing-farm.io/v0.1/requests/ab18144c-777b-4b54-aaea-a8f32f1c3d59)

Reply:
Hm, do we need to parse this? This seems to me as a detailed breakdown of all the testcases that were executed. Do we provide this information to our users or we just set the flags based on the overall status?

(maybe I'm missing something, so please gently point me in the right direction ;) )

Reply:
Imagine your tests failed and you need to know why, but all you get when clicking on the check link is [this](https://api.dev.testing-farm.io/v0.1/requests/ab18144c-777b-4b54-aaea-a8f32f1c3d59).
How user friendly would you consider it?

On the other hand, Tomas said that @thrix said ;-) that he will add this xunit -> HTML conversion on their side so we probably actually don't need to do this.

Reply:
we at least need to set the commit status to be passed/failed right? not sure if this info is present only in the xunit or also in some other metadata

would be awesome if TFT had some pretty view for the test results

Reply:
status is in ["result"]["overall"], we don't have to parse the xunit for that

Reply:
summary of IRC discussion:
jpopelka: we parse the xunit, but don't use the parsed data anywhere atm. It's because I thought I would need to create an HTML from of it at some point, but then Tomas said Miro will do that on TF side.
csomh: what about using http://artifacts.dev.testing-farm.io/ab18144c-777b-4b54-aaea-a8f32f1c3d59/work-fullQMe0r3
as it has all the logs but is pretty complicated
jpopelka: it's IMHO even more complicated than finding the links in the xunit. I had to go back to the xunit to see how to go through the directory structure to find the logs.